pipeline/frontend/yaml/compiler/convert.go

```diff
@@ -55,6 +55,18 @@ func (c *Compiler) createProcess(name string, container *yaml_types.Container, s
 		})
 	}
 
+	extraHosts := make([]backend_types.HostAlias, 0, len(container.ExtraHosts))
```

Comment:
```suggestion
	extraHosts := make([]backend_types.HostAlias, len(container.ExtraHosts))
```

And then just extraHosts[i].Name = name and same for IP

This way array objects will be allocated at once thus making it more optimal